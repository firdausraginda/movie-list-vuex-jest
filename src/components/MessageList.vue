<template>
  <div class="message-list">
    <ul>
      <li
        v-for="message in messagesProp"
        :key="message"
        class="message-list-ul-li"
      >
        {{ message }}
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  data() {
    return {};
  },
  props: {
    messagesProp: {
      type: Array,
      default: null
    }
  }
};
</script>

<style lang="scss" scoped>
.message-list {
  .message-list-ul-li {
    color: #42b983;
  }
}
</style>