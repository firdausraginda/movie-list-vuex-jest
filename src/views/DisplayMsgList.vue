<template>
  <div class="display-message-list">
    <MessageList :messagesProp="message" />
  </div>
</template>

<script>
import MessageList from "../components/MessageList";

export default {
  data() {
    return {
      message: ["Hey John", "Howdy Paco"]
    };
  },
  components: {
    MessageList
  }
};
</script>

<style lang="scss" scoped>
.display-message-list{
  background-color: lightblue;
}
</style>

