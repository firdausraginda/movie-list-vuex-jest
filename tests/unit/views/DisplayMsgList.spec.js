import { mount } from '@vue/test-utils'
import DisplayMsgList from '../../../src/views/DisplayMsgList.vue'
import MessageList from '../../../src/components/MessageList.vue'

describe('DisplayMsgList.spec.js', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = mount(DisplayMsgList)
        wrapper.setData({ message: ['Cat','Dog'] });
    });

    test('is DisplayMsgList component', () => {
        expect(wrapper.is(DisplayMsgList)).toBe(true)
        expect(wrapper.is('.display-message-list')).toBe(true)
    })

    test('message property has array type of data', () => {
        expect(typeof wrapper.vm.message).toBe('object')
    })

    test('has received ["Cat", "Dog"] as the message property', () => {
        expect(wrapper.vm.message).toEqual(["Cat", "Dog"])
    })

    test('has the expected html structure', () => {
        expect(wrapper.element).toMatchSnapshot()
    })

    test('contains a MessageList component', () => {
        expect(wrapper.contains(MessageList)).toBe(true)
        expect(wrapper.contains('.message-list')).toBe(true)
    })

    test('MessageList element exists and not empty', () => {
        expect(wrapper.find(MessageList).exists()).toBe(true)
        expect(wrapper.find(MessageList).isEmpty()).toBe(false)
    })

    test('MessageList has a "messagesProp" property equals to ["Cat", "Dog]', () => {
        expect(wrapper.find(MessageList).vm.messagesProp).toEqual(["Cat", "Dog"])
    })

})