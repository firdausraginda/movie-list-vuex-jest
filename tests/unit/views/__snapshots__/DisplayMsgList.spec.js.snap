// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayMsgList.spec.js has the expected html structure 1`] = `
<div
  class="display-message-list"
>
  <div
    class="message-list"
  >
    <ul>
      <li
        class="message-list-ul-li"
      >
        
      Cat
    
      </li>
    </ul>
  </div>
</div>
`;
